### ISSUE TITLE
Can't seem to target error message

### ISSUE THREAD

**Issue by MaartenWaegeman** [verbb/formie#1128]
### Describe the bug

I'm trying to target the error message of a (select / input) field to style it with tailwind. I am using the Tailwind theme config, which I've embedded in my formie.php config file (included in this request). My html looks like this:

<img width="789" alt="image" src="https://user-images.githubusercontent.com/7084691/196215860-e9035d8c-c231-4184-bca6-534281e4e6a7.png">

In my formie config, the value resetClasses is active, and I've tried to target the field with the values referenced in the documentation (fieldErrors / fieldError) but they don't seem to work. As the fui- class isn't removed, and the styling doesn't get applied.

```
config/formie.php:

<?php

return [
    // Default Formie config for Tailwind, see https://verbb.io/craft-plugins/formie/docs/theming/theme-config#ready-made-configs
    // https://github.com/verbb/formie-theme-configs/blob/main/tailwind/index.html
    // remember to add this file to tailwind config!!!

    'themeConfig' => [
        'resetClasses' => true,

        'formTitle'  => [
            'attributes' => [
                'class' => 'text-lg leading-6 font-medium text-gray-900',
            ],
        ],

        'alertError'  => [
            'attributes' => [
                'class' => 'rounded-md bg-red-50 border border-red-200 text-sm text-red-700 p-4 mb-4',
            ],
        ],

        'alertSuccess'  => [
            'attributes' => [
                'class' => 'rounded-md bg-green-50 border border-green-200 text-sm text-green-700 p-4 mb-4',
            ],
        ],

        'buttonWrapper'  => [
            'attributes' => [
                'class' => [
                    'flex justify-start -mx-2',
                    "{{ buttonsPosition == 'right' ? 'justify-end' }}",
                    "{{ buttonsPosition == 'center' ? 'justify-center' }}",
                    "{{ buttonsPosition == 'left-right' ? 'justify-between' }}",
                    "{{ buttonsPosition == 'right-save-left' ? 'justify-start flex-row-reverse' }}",
                    "{{ buttonsPosition == 'center-save-left' ? 'justify-center flex-row-reverse' }}",
                    "{{ buttonsPosition == 'center-save-right' ? 'justify-center' }}",
                    "{{ buttonsPosition == 'save-right' ? 'justify-between' }}",
                    "{{ buttonsPosition == 'save-left' ? 'justify-between flex-row-reverse' }}",
                ],
            ],
        ],

        'buttonContainer'  => [
            'attributes' => [
                'class' => 'flex justify-start',
            ],
        ],

        'saveButton'  => [
            'attributes' => [
                'class' => [
                    'inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md text-indigo-600 hover:underline focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500',
                    'mx-2',
                ],
            ],
        ],

        'backButton'  => [
            'attributes' => [
                'class' => [
                    'inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md text-slate-700 bg-slate-200 hover:bg-slate-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-slate-500',
                    'mx-2',
                ],
            ],
        ],

        'submitButton'  => [
            'attributes' => [
                'class' => [
                    'inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500',
                    'mx-2 order-10',
                ],
            ],
        ],

        'page'  => [
            'attributes' => [
                'class' => [
//                    currently gives issues
//                    "{{ form.hasMultiplePages() and page.id != currentPage.id ? 'hidden' : false }}",
                ],
            ],
        ],

        'pageTabs'  => [
            'attributes' => [
                'class' => 'border-b border-gray-200 -mb-px flex space-x-8 mb-6',
            ],
        ],

        'pageTab'  => [
            'attributes' => [
                'class' => 'py-4 px-1',
            ],
        ],

        'pageTabLink'  => [
            'attributes' => [
                'class' => [
                    'whitespace-nowrap py-4 px-1 border-b-2 font-medium text-sm',
                    "{{ (page.id == currentPage.id) ? 'border-indigo-500 text-indigo-600' : 'border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300' }}",
                    "{{ page.getFieldErrors(submission) ? 'text-red-500' : false }}",
                ],
            ],
        ],

        'pageTitle'  => [
            'attributes' => [
                'class' => 'text-lg leading-6 font-medium text-gray-900 mb-4',
            ],
        ],

        'progress'  => [
            'attributes' => [
                'class' => 'flex h-[1.2rem] text-sm text-white bg-slate-200 mt-4',
            ],
        ],

        'progressContainer'  => [
            'attributes' => [
                'class' => 'flex flex-col justify-center text-center font-medium bg-indigo-600',
            ],
        ],

        'subFieldRow'  => [
            'attributes' => [
                'class' => 'flex flex-wrap -mx-4 mb-4 last:mb-0',
            ],
        ],

        'nestedFieldRows'  => [
            'attributes' => [
                'class' => 'border py-4',
            ],
        ],

        'nestedFieldRow'  => [
            'resetClass' => true,
            'attributes' => [
                'class' => 'flex flex-wrap mb-4 last:mb-0',
            ],
        ],

        'row'  => [
            'resetClass' => true,
            'attributes' => [
                'class' => [
                    'flex flex-wrap -mx-4 mb-6',
                    "{{ not craft.formie.getVisibleFields(row) ? 'hidden' }}",
                ],
            ],
        ],

        'fieldContainer' => [
            'resetClass' => true,
            'attributes' => [
                'class' => 'md:flex w-full items-center',
            ]
        ],

        'fieldInputContainer' => [
            'resetClass' => true,
            'attributes' => [
                'class' => 'md:flex-1'
            ],
        ],

        'field'  => [
            'resetClass' => true,
            'attributes' => [
                'class' => 'px-4 flex-1',
            ],
        ],

        'fieldLabel'  => [
            'resetClass' => true,
            'attributes' => [
                'class' => 'sm:mb-1 md:mb-0 block text-sm font-medium text-gray-700 md:flex-1',
            ],
        ],

        'fieldInstructions'  => [
            'attributes' => [
                'class' => 'mb-1 text-sm text-gray-500',
            ],
        ],

        'fieldInput'  => [
            'attributes' => [
                'class' => [
                    'shadow-sm focus:ring-indigo-500 focus:border-indigo-500 block w-full sm:text-sm border-gray-300 rounded-md',
                    "{{ (submission.getErrors(field.handle) ?? null) ? 'border-red-300' }}",
                ],
            ],
        ],

        'fieldErrors' => [
            'attributes' => [
                'class' => 'mb-0',
            ],
        ],


        'fieldError'  => [
            'attributes' => [
                'class' => 'text-red-500 text-sm mt-1',
            ],
        ],

        'errorMessage' => [
            'attributes' => [
                'class' => 'text-blue-500 text-sm mt-1',
            ],
        ],

        'fieldAddButton'  => [
            'attributes' => [
                'class' => 'inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md text-slate-700 bg-slate-200 hover:bg-slate-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-slate-500',
            ],
        ],

        'fieldRemoveButton'  => [
            'attributes' => [
                'class' => 'inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md text-slate-700 bg-slate-200 hover:bg-slate-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-slate-500',
            ],
        ],

        'fieldLimit'  => [
            'attributes' => [
                'class' => 'mt-2 text-sm text-gray-500',
            ],
        ],

        'agree'  => [
            'fieldOption' => [
                'attributes' => [
                    'class' => 'flex items-start mb-2',
                ],
            ],
            'fieldOptionLabel' => [
                'attributes' => [
                    'class' => 'ml-2 text-sm leading-4',
                ],
            ],
            'fieldInput' => [
                'resetClass' => true,
                'attributes' => [
                    'class' => 'focus:ring-indigo-500 h-4 w-4 text-indigo-600 border-gray-300 rounded',
                ],
            ],

        ],

        'checkboxes'  => [
            'fieldOption' => [
                'attributes' => [
                    'class' => 'flex items-start mb-2',
                ],
            ],
            'fieldOptionLabel' => [
                'attributes' => [
                    'class' => 'ml-2 text-sm leading-4',
                ],
            ],
            'fieldInput' => [
                'resetClass' => true,
                'attributes' => [
                    'class' => 'focus:ring-indigo-500 h-4 w-4 text-indigo-600 border-gray-300 rounded',
                ],
            ],

        ],

        'dropdown'  => [
            'fieldInput' => [
                'resetClass' => true,
                'attributes' => [
                    'class' => 'shadow-sm focus:ring-indigo-500 focus:border-indigo-500 block w-full sm:text-sm border-gray-300 rounded-md',
                ],
            ],
        ],

        'hiddenField'  => [
            'field' => [
                'resetClass' => true,
                'attributes' => [
                    'class' => 'hidden',
                ],
            ],
        ],

        'radioButtons'  => [
            'fieldOption' => [
                'attributes' => [
                    'class' => 'flex items-start mb-2',
                ],
            ],
            'fieldOptionLabel' => [
                'attributes' => [
                    'class' => 'ml-2 text-sm leading-4',
                ],
            ],
            'fieldInput' => [
                'resetClass' => true,
                'attributes' => [
                    'class' => 'focus:ring-indigo-500 h-4 w-4 text-indigo-600 border-gray-300',
                ],
            ],
        ],

        'recipients'  => [
            'field' => [
                'attributes' => [
                    'class' => "{{ field.getIsHidden() ? 'hidden' : false }}",
                ],
            ],
        ],

        'repeater'  => [
            'nestedField' => [
                'attributes' => [
                    'class' => 'relative'
                ],
            ],
            'nestedFieldWrapper' => [
                'attributes' => [
                    'class' => 'mb-4',
                ],
            ],
            'fieldRemoveButton' => [
                'attributes' => [
                    'class' => 'absolute top-0 right-0 m-1 inline-flex items-center px-2 py-1.5 border border-transparent text-xs font-medium rounded-md text-slate-700 bg-slate-200 hover:bg-slate-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-slate-500',
                ],
            ],
        ],

        'signature'  => [
            'fieldInputContainer' => [
                'attributes' => [
                    'class' => 'relative',
                ],
            ],
            'fieldCanvas' => [
                'attributes' => [
                    'class' => 'w-full h-[8rem] border bg-slate-100',
                ],
            ],
            'fieldRemoveButton' => [
                'attributes' => [
                    'class' => 'absolute top-0 right-0 m-1 inline-flex items-center px-2 py-1 border border-slate-300 text-xs font-medium rounded-md text-slate-700 bg-slate-200 hover:bg-slate-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-slate-500',
                ],
            ],

        ],

        'table'  => [
            'fieldTable' => [
                'attributes' => [
                    'class' => 'min-w-full divide-y divide-gray-300',
                ],
            ],
            'fieldTableHeaderColumn' => [
                'attributes' => [
                    'class' => 'px-3 py-3.5 text-left text-sm font-semibold text-gray-900',
                ],
            ],
            'fieldTableBody' => [
                'attributes' => [
                    'class' => 'divide-y divide-gray-200 bg-white',
                ],
            ],
            'fieldTableBodyColumn' => [
                'attributes' => [
                    'class' => 'whitespace-nowrap px-3 py-4 text-sm text-gray-500',
                ],
            ],
        ],

    ]
];
```

### Steps to reproduce

1. 
2. 

### Form settings

- Multi-page form: Yes 
- Submission Method: Page Reload
- Client-side Validation: No
- Custom Form Templates: No

### Craft CMS version

4.2.7

### Plugin version

2.0.12

### Multi-site?

No

**Comment by engram-design** [comment]
I noticed you mentioned you're **not** using client side validation ("Validate Form on Submit"), but you must be in order to get this error markup.

For now, the client-side validation (which uses [bouncer.js](https://github.com/cferdinandi/bouncer)) does produce different markup compared to server-side rendered errors (which is one of the main reasons we'll be switching to something else).

For example:

Server-side errors:
```
<ul>
    <li class="text-red-500 text-sm mt-1">Testing cannot be blank.</li>
</ul>
```

Client-side errors:
```
<div class="fui-error-message" id="fui-error-fields[test]">Please fill out this field.</div>
```

And you'll notice even the error text is different, so lot's to cleanup and standardise.

But, for now, I've fixed this lack of class reference for the next release. It should now correctly reference the `fieldError` classes you add to theme config. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.12",
  "...": "..."
}
```

Then run `composer update`.
